cardano-api/src/Cardano/Api/Tx.hs

```diff
@@ -503,19 +504,21 @@ getTxWitnesses (ShelleyTx era tx) =
         map (ShelleyBootstrapWitness era) (Set.elems bootWits)
      ++ map (ShelleyKeyWitness       era) (Set.elems addrWits)
 
-
 makeSignedTransaction :: forall era.
-                         [KeyWitness era]
-                      -> TxBody era
-                      -> Tx era
-makeSignedTransaction witnesses (ByronTxBody txbody) =
+      Alonzo.IsValid
```

Comment:
So I know this is where the spec specifies the `IsValid` flag but if we specify this here, we then have to provide an `IsValid` value for Byron txs that doesn't get used. What we should do is add the `IsValid` flag in `ShelleyTxBody` instead to avoid this.

Reply:
The flag has been moved to `ShelleyTxBody`.